### ISSUE TITLE
Create drasil-gen SubPackage

### ISSUE THREAD

**Issue by samm82** [JacquesCarette/Drasil#767]
The `drasil-gen` subpackage should be created with the code that drives the generation

## TODO
- [ ] Move Language.Drasil.Generate

**Comment by JacquesCarette** [comment]
This should contain the routines that drive the generation. L.D.Generate for sure. Eventually, I think we'll want to split off some pieces currently in drasil-code (and drasil-doc ?) to put them here.  Right now, this sub-package might be tiny.  That's fine.

**Comment by samm82** [comment]
Should it re-export everything or not? @JacquesCarette

**Comment by JacquesCarette** [comment]
I think it just needs to export `gen` (for now). But we'll see.

**Comment by samm82** [comment]
I think one example also requires `genCode`

**Comment by JacquesCarette** [comment]
But isn't `genCode` in `drasil-code`? [If not, then `genCode` should be exposed. It can stay here]

**Comment by samm82** [comment]
`genCode` is in L.D.Generate

**Comment by elwazana** [comment]
@JacquesCarette @szymczdm Hello Dr. Carette and Dan, I've pulled out Generate.hs from `drasil-code` and created `drasil-gen` to house it, I've also fixed all the related import statements. But now I'm hitting this error that I'm not sure how to fix. 

![image](https://user-images.githubusercontent.com/35268928/42644927-ca49044e-85ca-11e8-951d-14722917d6be.png)

I've asked @samm82  for help but he is also unsure as to what the cause is.
